lib/composite_primary_keys/connection_adapters/abstract/database_statements.rb

```diff
@@ -32,6 +31,16 @@ def insert(arel, name = nil, pk = nil, id_value = nil, sequence_name = nil, bind
           last_inserted_id(value)
         end
       end
+
+      private
+
+      def active_record_result(value)
+        return value if @config[:adapter] != "trilogy"
+
+        # Create an AR Result so we can respond to #empty?
+        # since Trilogy:Result doesn't respond to it.
+        ActiveRecord::Result.new(value.fields, value.to_a)
```

Comment:
Hmm, this seems like a pretty aggressive change that could have unforeseen consequences. Maybe add `empty?` to the Trilogy::Result instead? Or if trilogy just skip the check?

Reply:
@cfis are you suggesting adding #empty? to the trilogy gem? If so, I actually opened a PR to do that, however the maintainers didn't see it necessary as it's not meant to behave like a AR Result.

Reply:
I was thinking of adding something like this below line 8:
```ruby
return value if @config[:adapter] == "trilogy"
```

But open for suggestions on how you'd best handle this.